fs/wildcard.py

```diff
@@ -123,55 +135,92 @@ def get_matcher(patterns, case_sensitive):
     if not patterns:
         return lambda name: True
     if case_sensitive:
-        return partial(match_any, patterns)
+        return lambda name: match_any(patterns, name, accept_prefix)
     else:
-        return partial(imatch_any, patterns)
+        return lambda name: imatch_any(patterns, name, accept_prefix)
+
+
+class _PatternMatcher:
+    def __init__(self, pattern, case_sensitive, accept_prefixes):
+        self.case_sensitive = case_sensitive
+        self.accept_prefixes = accept_prefixes
+
```

Comment:
I'd also store the pattern for good measure as an attribute here, it could help debugging.

Reply:
Also, this would allow writing a roundtripping `__repr__` implementation:

```python
from .repr import make_repr

class _PatternMatcher:
    def __repr__(self):
        return make_repr(
            '_PatternMatcher', 
            self.pattern, 
            case_sensitive=(self.case_sentitive, None), 
            accept_prefixes=(self.accept_prefixes, None)
        )
```